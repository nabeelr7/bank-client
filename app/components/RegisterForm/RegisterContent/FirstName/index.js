import React, { useState, useEffect } from 'react';
import { useSelector, useDispatch } from 'react-redux';
import { createStructuredSelector } from 'reselect';
import { Input } from 'antd';
import { makeSelectFirstName } from 'containers/RegisterPage/selectors';
import { changeInputAction } from 'containers/RegisterPage/actions';
// import { FormattedMessage } from 'react-intl';
import { StyledFormItem } from '../../RegisterForm.style';
// import messages from './messages';

const stateSelector = createStructuredSelector({
  firstName: makeSelectFirstName(),
});

export default function FirstName() {
  const { firstName } = useSelector(stateSelector);
  const dispatch = useDispatch();
  const [isStartedRegistration, setIsStartedRegistration] = useState(false);
  const isName = /^[a-z ,.'-]+$/i;

  const checkStringConsistsLettersOnly = (_, value) => {
    if (value && !isName.test(value)) {
      return Promise.reject(new Error('The name entered must be valid.'));
    }

    if (!value && isStartedRegistration) {
      return Promise.reject(new Error('First name is required.'));
    }

    return Promise.resolve();
  };

  useEffect(() => {
    setIsStartedRegistration(true);
  }, [firstName]);

  return (
    <StyledFormItem
      label="First name"
      name="firstName"
      rules={[{ validator: checkStringConsistsLettersOnly }]}
    >
      <Input
        onChange={(event) => dispatch(changeInputAction(event.target))}
        name="firstName"
        value={firstName}
        placeholder="Enter first name"
      />
    </StyledFormItem>
  );
}
