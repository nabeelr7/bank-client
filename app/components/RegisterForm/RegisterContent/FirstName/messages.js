/*
 * FirstName Messages
 *
 * This contains all the text for the FirstName component.
 */

import { defineMessages } from 'react-intl';

export const scope = 'app.components.RegisterForm.RegisterContent.FirstName';

export default defineMessages({
  label: {
    id: `${scope}.label`,
    defaultMessage: 'First name',
  },
  validation_valid: {
    id: `${scope}.validation_valid`,
    defaultMessage: 'The first name entered must be valid.',
  },
  validation_required: {
    id: `${scope}.validation_required`,
    defaultMessage: 'First name is required.',
  },
  placeholder: {
    id: `${scope}.placeholder`,
    defaultMessage: 'Enter first name',
  },
});
