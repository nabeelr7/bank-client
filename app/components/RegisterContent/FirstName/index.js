import React, { useState, useEffect } from 'react';
import { useSelector, useDispatch } from 'react-redux';
import { createStructuredSelector } from 'reselect';
import { Input } from 'antd';
import { makeSelectFirstName } from 'containers/RegisterPage/selectors';
import { changeInputAction } from 'containers/App/actions';
import { intlShape, injectIntl } from 'react-intl';
import { StyledFormItem } from 'components/Form/Form.style';
import { nameValidation, disabledSpacesInput } from 'helpers';
import messages from './messages';

const stateSelector = createStructuredSelector({
  firstName: makeSelectFirstName(),
});

function FirstName({ intl }) {
  const { firstName } = useSelector(stateSelector);
  const dispatch = useDispatch();
  const [isStartedRegistration, setIsStartedRegistration] = useState(false);

  const onChangeInput = (event) => dispatch(changeInputAction(event.target));

  const checkStringConsistsLettersOnly = (_, value) => {
    if (value && !nameValidation(value)) {
      return Promise.reject(
        new Error(intl.formatMessage(messages.validationValid)),
      );
    }

    if (!value && isStartedRegistration) {
      return Promise.reject(
        new Error(intl.formatMessage(messages.validationRequired)),
      );
    }

    return Promise.resolve();
  };

  useEffect(() => {
    setIsStartedRegistration(true);
  }, [firstName]);

  return (
    <StyledFormItem
      label={intl.formatMessage(messages.label)}
      name="firstName"
      rules={[
        { validator: checkStringConsistsLettersOnly },
        { whitespace: false },
      ]}
    >
      <Input
        onKeyPress={disabledSpacesInput}
        onChange={(event) => onChangeInput(event)}
        name="firstName"
        value={firstName}
        placeholder={intl.formatMessage(messages.placeholder)}
      />
    </StyledFormItem>
  );
}

FirstName.propTypes = {
  intl: intlShape.isRequired,
};

export default injectIntl(FirstName);
