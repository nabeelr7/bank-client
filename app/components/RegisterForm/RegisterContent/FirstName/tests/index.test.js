import React from 'react';
import { render } from '@testing-library/react';
import { createMemoryHistory } from 'history';
import { Provider } from 'react-redux';
import { IntlProvider } from 'react-intl';
import { DEFAULT_LOCALE } from 'locales';
import configureStore from 'configureStore';
import FirstName from '../index';
import './__mocks__/matchMedia';

describe('<FirstName />', () => {
  const history = createMemoryHistory();
  const store = configureStore({}, history);

  it('should render a information', () => {
    const { container } = render(
      <Provider store={store}>
        <IntlProvider locale={DEFAULT_LOCALE}>
          <FirstName />
        </IntlProvider>
      </Provider>,
    );
    expect(container.firstChild).toMatchSnapshot();
  });
});
