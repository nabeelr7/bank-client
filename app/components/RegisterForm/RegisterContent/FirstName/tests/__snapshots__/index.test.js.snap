// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FirstName /> should render a information 1`] = `
<div
  class="ant-row ant-form-item RegisterFormstyle__StyledFormItem-sc-1yq7sv9-7 iamOBg"
>
  <div
    class="ant-col ant-form-item-label"
  >
    <label
      class=""
      for="firstName"
      title="First name"
    >
      First name
    </label>
  </div>
  <div
    class="ant-col ant-form-item-control"
  >
    <div
      class="ant-form-item-control-input"
    >
      <div
        class="ant-form-item-control-input-content"
      >
        <input
          class="ant-input"
          id="firstName"
          name="firstName"
          placeholder="Enter first name"
          type="text"
          value=""
        />
      </div>
    </div>
  </div>
</div>
`;
